include/battle_ai_util.h

```diff
@@ -37,6 +37,8 @@ static inline bool32 IsMoveUnusable(u32 moveIndex, u32 move, u32 moveLimitations
         || moveLimitations & 1u << moveIndex;
 }
 
+typedef u32 (*MoveFlag)(u32 getFlag);
```

Comment:
```suggestion
typedef bool32 (*MoveFlag)(u32 battler);
```
I _think_ it being a flag implies `bool32`? And all the existing functions took their argument as a `u32 battler`, so I'm guessing that name would be appropriate?

Reply:
oh oops, I changed the func signature to a `bool32` but didn't apply it here.

Though I'm not sure if the battler argument is correct?
```
bool32 HasMoveWithFlag(u32 battler, MoveFlag getFlag)
```

Reply:
Oh you're right, `getFlag` gets called on a `u32 move`!